pkg/baseboard/baseboard.go

```diff
@@ -57,7 +57,18 @@ func (i *Info) String() string {
 // New returns a pointer to an Info struct containing information about the
 // host's baseboard
 func New(opts ...*option.Option) (*Info, error) {
-	ctx := context.New(opts...)
+	return newWithContext(context.New(opts...))
+}
+
+// New returns a pointer to an Info struct containing information about the
+// host's baseboard, reusing a given context.
+// Use this function when you want to consume this package from another,
+// ensuring the two see a coherent set of resources.
+func NewWithContext(ctx *context.Context) (*Info, error) {
+	return newWithContext(context.FromParent(ctx))
```

Comment:
I don't see a need for the `context.FromParent` function or to embed one context in another. Why not just set the context you receive directly into the Info struct? In other words, just do this (shown for the CPU package...)?

```go
// New returns a pointer to an Info struct that contains information about the
// CPUs on the host system
func New(opts ...*option.Option) (*Info, error) {
    return NewWithContext(context.New(opts...))
}

// NewWithContext returns a pointer to an Info struct containing information
// about the CPUs on the host system given an existing Context 
func NewWithContext(ctx *context.Context) (*Info, error) {
    info := &amp;Info{ctx: ctx}
    if err := ctx.Do(info.load); err != nil {
        return nil, err
    }
    return info, nil
}
```

Reply:
This is a very good point. Thinking about it, we actually don't. What we actually need, is a mechanism for shared contexts to know when Setup() and Teardown() must happen (= the outermost package must do that, the dependent contexts must not).
So we need a form of reference counting. Let me update the code.